manuskript/ui/editors/editorWidget.py

```diff
@@ -313,17 +312,26 @@ def setCurrentModelIndex(self, index=None):
         if self._model:
             self.setView()
 
-    def updateIndexFromID(self):
+    def updateIndexFromID(self, fallback=None, ignore=None):
         """
         Index might have changed (through drag an drop), so we keep current
         item's ID and update index. Item might have been deleted too.
+
+        It will ignore the passed model item to avoid ambiguity during times
+        of inconsistent state.
         """
-        idx = self._model.getIndexByID(self.currentID)
 
-        # If we have an ID but the ID does not exist, it has been deleted
+        idx = self._model.getIndexByID(self.currentID, ignore=ignore)
+
+        # If we have an ID but the ID does not exist, it has been deleted.
         if self.currentID and idx == QModelIndex():
-            # Item has been deleted, we open the parent instead
-            self.setCurrentModelIndex(self.currentIndex.parent())
+            # If we are given a fallback item to display, do so.
+            if fallback:
+                self.setCurrentModelIndex(fallback)
+            else:
```

Comment:
You don't need the 'if fallback' or the else part, because fallback will always be set to the parent when rows are being removed, and in the insertion case, your current ID will never end up deleted. I'd just put 'fallback' as a non optional argument in this function too.

Reply:
The intent for the fallback parameter is to specify an alternative view in the case you do not want the default action of the tab being closed. As such the design for that parameter is optional, and changing the function signature so that it is required communicates that the desired behaviour for this function is to always show the fallback when updating the ID. It isn't: existing code clearly shows that closing a tab is the preferred action.

From our discussion last night I realize we are of different minds here; you are an advocate of 'show something else instead' and not very much into the entire 'close the tab' business. :-) However, the original design for the code is to close the tab if the item is removed; the 'show the parent' logic is something this PR adds which definitely has its uses... but beyond the 'existing code says X' issue, I do not believe it is the desired default solution.

For example, take a folder with five texts in it. These five texts are open in the editor on separate tabs because I am working on them. Then I decide I don't want the folder there anymore, so I 'cut' it. The result with the current code will be to have five tabs that all show the root. To me as a user, all those root windows are utterly useless and have nothing to do with what I was doing in them anymore, meaning I will have to close those tabs manually afterwards. IMHO, the fallback option should only be used when you really want to keep the tab open.

My goal here is not to change the default behaviour or intended operating principles of Manuskript, but to fix a bug. The original code (that doesn't quite work) shows that we want to close tabs here, so I structure the code with that intent in mind. The best way to do that right now is by showing the parent instead in all situations where a fallback is needed. But that is not a long term solution. Truly fixing all this is out of scope for this PR, but it will eventually get fixed.

Note that the code in its current form should not trigger the else condition because a fallback is always specified, so I do not see any functional problems with the code as it is. I am fine with changing some implementation details to make this PR neater, but I am quite stubborn on preserving the current function signature and meaning of the fallback parameter.

Reply:
Alright, that sounds good to me. It's a good explanation too, and I agreed with you in our last discussion that it's a difference of opinion, nothing inherently wrong either behaviors, but I don't think it's "important enough" to have that behavior be a settings option for the user. That's why I suggested to open the discussion here, see what @gedakc or @olivierkes thinks. As for "original intent" or "existing code", I disagree. That was the old commented out code, it was maybe the initial implementation that was later removed for a better implementation (which was to not close the tab), but maybe it was a different opinion from a different developer, or maybe it was removed because it kept crashing and they changed the index instead as a temporary fix, or maybe closing a tab was there as an early implementation but then they changed it to something they thought was a better behavior.. I don't know, which is why discussing it would be good, but I just don't accept your argument of "existing code shows that closing a tab is the preferred action."

Either way, while I prefer the tab not to close but switch to the parent, I have no issues with whatever choice is decided. In my case anyway, I usually work with a single tab.

Reply:
When writing I usually work with a single tab too, so I don't have a strong opinion on which way this decision lands.  We can change it later if it becomes a problem.

Regarding original intentions, I don't think we will always know what these were/are.  @olivierkes has not been active in the project for over a year so I don't think we should hold up this PR for @olivierkes' input.